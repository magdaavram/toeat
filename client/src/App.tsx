import React, { useState } from 'react';
import styled from 'styled-components';
import Header from 'components/Header';
import Routes from 'routes';
import { Router } from 'react-router-dom';
import history from 'services/history';
import RecipeContext from './RecipeContext';

const Container = styled.div`
  padding: 45px;
`;

const Main = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 126px;
`;

const App = () => {
  const [id, setId] = useState('');
  const handleIdChange = (id: string) => setId(id);

  return (
    <Container>
      <Router history={history}>
        <RecipeContext.Provider value={{ id: id, setId: handleIdChange }}>
          <Header />
        </RecipeContext.Provider>

        <Main>
          <RecipeContext.Provider value={{ id: id, setId: handleIdChange }}>
            <Routes />
          </RecipeContext.Provider>
        </Main>
      </Router>
    </Container>
  );
};

export default App;
