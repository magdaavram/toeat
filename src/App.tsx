import React from 'react';
import styled from 'styled-components';
import Header from 'components/Header';
import Routes from 'routes';
import { Router } from 'react-router-dom';
import history from 'services/history';

const Container = styled.div`
  padding: 45px;
`;

const Main = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 126px;
`;

const App = () => {
  return (
    <Container>
      <Router history={history}>
        <Header />
        <Main id="main">
          <Routes />
        </Main>
      </Router>
    </Container>
  );
};

export default App;
