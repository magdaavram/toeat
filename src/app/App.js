import React from 'react';
import logo from './images/logo.png';
import styled from 'styled-components';
import './style/';

const Page = styled.div`
  min-height: 100vh;
  background-color: var(--color--beige);
  color: var(--color--dark-purple);
  font-family: 'Roboto Regular';
  font-size: var(--font-size--regular);
  padding: 0 45px;
`;

const App = () => {
  return (
    <Page>
      <header>
        <img src={logo} alt="App logo" />

        <p>
          This is just the beginning.
        </p>
      </header>
    </Page>
  );
}

export default App;
