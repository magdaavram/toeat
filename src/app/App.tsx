import React from 'react';
import styled from 'styled-components';
import Header from './components/Header';
import Routes from './routes';
import { Router } from 'react-router-dom';
import history from './services/history';

const Container = styled.div`
  padding: 45px;
`;

const Main = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 126px;
`;

const App = () => {
  return (
    <Container>
      <Header />
      <Main>
        <Router history={history}>
          <Routes />
        </Router>
      </Main>
    </Container>
  );
};

export default App;
