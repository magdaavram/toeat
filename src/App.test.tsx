import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from 'App';

it('renders app without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

it('renders router, header and main components', () => {
  const wrapper = shallow(<App />);
  expect(wrapper.find('Router')).toHaveLength(1);
  expect(wrapper.find('Header')).toHaveLength(1);
  expect(wrapper.find('#main').find('Routes')).toHaveLength(1);
});
