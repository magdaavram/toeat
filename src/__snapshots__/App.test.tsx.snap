// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render app correctly 1`] = `
<div
  className="sc-fznyYp dQMOKJ"
>
  <div
    className="sc-fzoLsD lgjcow"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <svg
        height="50"
        viewBox="0 0 200 50"
        width="200"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <image
            height="48"
            href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAMQAAAAwCAYAAABOpdVkAAAAAXNSR0IB2cksfwAACYNJREFUeJztXWmIHEUUXvBEvFBExRMVRPFE8Fb84y26CmpU1HjELOt0VWbaaiYMy7q0rR1HV11v44G3Lh54xBvEg+AVxQMlJjGDq5Bs8Fa8fuh7PTPr7ExVz1T3q7msD4pZlpn3vqp6X3VVdfXrgYEm8H1nK99zThwTlx+bzWY3afZ9C4tOwXisBrnMXF84KwLB/sHiC/YhfB5G7sjCIiVCwYeNxyoYvLPqYKa47GpSJxYWBAg89rrxWJU6gf+ROrGwIEBDnJqIVSsIi16BFYSFRQ2sICwsamAFYWFRAysIC4saWEFYWNSATBABY1v7npORBn+68qQvnPPSVDK68yjY+cgNPpeBzS+hrIG/f4v37fwEn1Oh4J/B51LfdSZCL3N0Gi5JUGBsxyCfvRT5hx5fBu2xEvhMQ/mjF/gjApEZBD7XQh3uAx5LAtd5DzithvJru7kbi1WsG8RZ2UkuM5dYCLPKaG5oV92Kjwl+HDToQ0D0L0oueCcTOuim8Xx+e4oOUgHa9CLg/hx1W7aLPx5/wEAGnz90E3fDsTodxSr88YRJQaD6tCot2B1G+UTF+XY8z05J2jFxCD32aK/ynyjwnQKXLelW7qZjFa7imYFQsPlGGwBU3XqF+bT5YKppAMFHk3RMTId916v8oe0PBzF83s3cTceqn3MOHgjdzD6V+R69E4+913qHsPF2BlO1jHv8QN2OUfCf7FX+fp6dDGL4vdu5G47V/xbg+fzFmxXz/BxQyJW1RbXLVP89WRnNOce0WtGilz2hBdLLwTdezh9A8cT5Dl1+s++yR6HxXoEpzLdxdrGBS6OjW+p0TD18j1/eCv9oUQpTSFykxrafYLdGAos2E5zvTfMHGw+3wH818Hq1Ivw74YoSyrgHHr8H7L0K3JfDiB4rsiTclbGqCPJWYrUIa9aWnLdr2xXsvqtuOH7FRIHtmMZ+WMjtDZ15u8oHBOrNiblzvmeTkWdhWv4jrrNvEHMFSsO/6A1tFxOwa0KRPaMoxLZJbHuRbe7FjeowYN2QlHstmo76JE7aIAgYyQ9VNNYUdPQhpL6Ec57CVympTRSsqrPHRGZ3QvoDypEwBf8x18krbH5MxXvOnDkbYODLRcc+ovDRN4IoevwaWWWKeZal9FMF2H5QOsp6zgmJ7Hn8DZk9nSmjDnAqSMtfvqc/ls8eRMm7VLpwQwx+eVvpb83Xo28EAfa+aOhc4XxF6aMWEwvdQxQj1e26ttTTDecpE9wR0W4QEX+E7KoDohun5o0IXH6atL00diKVtvtBEIVCYXN5RfhiKh8yBK7zaUNAeXyZrh1fZI6XjtYQZAZozwBE8YvE79Kk9mA6U7u//w4l11l+crltpO1FIMC+EAQudts53ahCMRef1rUDgXRmJ/jL+gUXrmlsYrBSTF2aAYL/a0mbTaa12xeCUI2wKBQqH1K/isVpad68jXTswJXskk7wD+Q7TlMmfVIBt8Ipr25V9IUglAEFoxWVDxlUglgshNauEC78O8K/fJ+ift3yE7WfhdnhvXw3Mwd3osJo25q/GMVE2mJAzH0hCFVgUtnX9as71ek2/hS2cXcJpoIF33XelwaZoZKWtxWEAb9UgkA7JoupdgtkaYesIGqcWEEkstOLQdXL3JX8rSDS+f2/CgLWB0O9yr0KKwgDfv+PgigKtn+neVtBVGAFQVn4oiRtAQvoq2LsTgZe9ijcRsaDengmKYmPVtosrV0rCAN+e1YQ2E95fmqStoDfN9y1N9kHVhAx6FdBmN5lqi14NDxpO+ADN4pAMpbQ2goiBv0qCEO0yYE33WT8w6y5DB9WEDHoNkEEmu8T6HVBqJ6FwCuHKZ89LYjokcf6xiJ6oAPRbUc3Jkb4zjp2OnV0gwqh4inC/PCwsTQ3BgXRmKCCWhCh4I80dLbHvqGy33WH+0ql9XTsdOpw34x/xrauriUS/R7PJsnagWA3Se2TXWdCEHh8vw2CkOVIcv4is99dx7/X6Nrp1PFvBPhZOttvZlDXBswAGga8SBAjI1uY4IzbtpUMjCauEM8YF0SUwtDgCKh8QAhGEQr7KoDQP6j3iR2la0d1hYPimuBdBQjxLIrO9135PQjfc/YwwRts5xTtlVoQ0hPArvPnPM0j/bFQJoaCqQKhj48lPpZT2a8HJqNSdIr2U1thbp70CbCA4IGXOPiyZ5MTCCJwMxdLBeGyQ6k5F8X8bYNyTlgjglDFKkylzqbgHwGTDSsq8BaVD0xTIhedmb3wSm4nSRAky/ABv31Bai9lomelP0UQJ3kuGad27bhCj2bn7wLt8bJKDCRXCEWswjr4JYo6zCBKOixxhMmiKOwXPXaAqpFGuN6uTzOMe85JCl/vJrUZk15xDU6pKPmjaAO3MSlDJEAIiCQ2A9nuDEGAVlEUWZxWrowTA5U/VaxSzmjichlF0wzMApHWB9h5R+UDpge3waiY6FgCojQ0tDHYuKDuQfrZPlIkBVgsxG6xne06d6fhj4gSreXZAqjHz4pR8KHEttUJ3NaNCuf0JDYhJo4I3Ci90FvNhEApiPhY5YtGYfCl8IPTjOfjK8R/qdzEm8TFDQbYiHD2a9l+a2nO1yZKZSnYn03sfpK2ffz4A3Iz/FtNZQnivRG+dz+MbM+2kss0zfRMdbe6pvxQzqfEn8agkqSQvB7a+rFK8K8Km7z7Aur2mClBIMDW283qUx+rQZ7vqeWkVCptCAb+blXtM0VjoVfJG6pnn6BQZQfExM6d4A9BWkzNXThvtoWry+41fXc/6XF2FIdmo7HDkjjSeUEGfH9VW8VAnD8JbE61VQxE+avi8ruStbVwbkFf7Tjugqn2E3CcxoFfy1HJ8zYNFKkgVQXn2HqVMf/0Fk5DqNLgN/D3WKEdYoBpyuOUvCcWOjuoUk2ma2v2B04pq34U0+PVlHVBYP/qps7XjdUZlOe5vIUFk/NEEvtRusZorq3et07YOR9h0t206eOboZjPnluec9O9kqrC/zdYpN8FV4YjTXGHPlsEbf9jer78KxCtX5/aB89J1ffrmMsKJuqC/Yz93ZrQk8XqLOB0qCiy52MqwqCcPAuPE1SnDZN4tCGtD9wZAPsLKncin6mcWWn2pqGpCpfJiBuMSu3ISiflX82S4Tp34554edSKXkwYx38tptuE776G58lwkT0m2PzBwcH128F55LLLtohedglBAuLDdDQroKwLpOtIjjtfy6N3MZSPglwXuPEPK0GdhisL22ns13YcdcH+xzhoFqv/AkAAfQdXB/plAAAAAElFTkSuQmCC"
            id="logo-icon"
            width="196"
          />
        </defs>
        <use
          href="#logo-icon"
          transform="matrix(1,0,0,1,2,1)"
        />
      </svg>
    </a>
    <div
      className="sc-AxmLO MKJBh"
    >
      <div>
        <a
          href="/add-recipe"
          onClick={[Function]}
        >
          <button
            className="sc-AxjAm ewqDcP sc-fzozJi Jilpr top-icon"
            onClick={[Function]}
            type="button"
          >
            <svg
              height="50"
              viewBox="0 0 50 50"
              width="50"
              xmlns="http://www.w3.org/2000/svg"
            >
              <defs>
                <image
                  height="50"
                  href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADIAAAAyCAMAAAAp4XiDAAAAAXNSR0IB2cksfwAAAPZQTFRFAAAASk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pgAA4OAAAAFJ0Uk5TAB9cnsnj9f4dg97/HCy1/RSp4MGaqBNX8uqXPwsMVgSbzUoDS84KScDMlgjL+To79jAvd3bP0Cu0guvdmB5AW53C9/R1yJxagadYVZm/ygni/BkwfscAAAJfSURBVHicnVbpQtpAEB4RSHE4TLgERCoQSou2iAjIpYhcglXf/2U6EwIkmw2Ezg/YndkvOzs3gINOfKf+QFBRggH/qe/EKRfpW+gMbXQWCu8FRKIxdFAsGnEFnKuKeUiLq4mEGtfMDyjJlByRvjDkmWzucsPK57IZg3lxJQEUvhuy62LJzi+VdUOQKIiISpz5gR+y64sBllUrAuInc3/V5CrXsiy9sWEKfMdtWg5gSt/yPVbd+B3B3+4IgD91fo/lG3yHgLhrCBi+Z2u3c7auoNU9YlPQjW298U+SXy4o8oCoCyy2QXK9jJDPA6KtWogtgVUjWyttYxkltMMfEggU6WCHF2EKpGtRKoUAxUGM4zpE2KI3SJmOPtI/5Uem5A1S6iL2KAcJmXUI5RDo0+EB+Og35xUypMNP8ExPuvQKyZM/nsGPqDllLhDQEP1A/ol7h4zI6xBEVM39/UNrSy+I9d1O38SbijiG2Dam75zFZUdmXCfo5RZIwyvEolhT36lStyv2alXsP55/tJEnhivzXiHsyik8HRswMxgcH5bQOzr44ZGQZW8QTrE5rBNZLCYHEhk6skzeWy6gfXxRMkqfaDTdqS2ba2GuU5IC20R8tXPerAUWriRlvCEr48vdfkXb+uFmsbIwCtUDLemN73i3tcvKDedR1q3x8cvx74edW6ka7dVZaWHTXt8/RH5hZSSsXnZp4itHEydarkeFbn+4zZ/8sN9djwpLCYAotTAHEkUb8UAy0jb7hctAQtTuyMaeTtsVwBSe9+yA3nz/cGXQYDadfI6/vsafk+ls4JT/Aw+fVYXLsMSIAAAAAElFTkSuQmCC"
                  id="add-icon"
                  width="50"
                />
              </defs>
              <use
                href="#add-icon"
                x="0"
                y="0"
              />
            </svg>
          </button>
        </a>
      </div>
    </div>
  </div>
  <div
    className="sc-fznLxA domtIr"
  >
    <form
      className="sc-fzplWN emtqnU"
      onSubmit={[Function]}
    >
      <input
        className="sc-fzpans sc-fznyAO ggHIiT"
        placeholder="Find recipes by ingredients or name..."
        type="search"
      />
      <button
        className="sc-AxjAm ewqDcP sc-fznKkj hmSzya"
        type="submit"
      >
        <svg
          height="27"
          viewBox="0 0 27 27"
          width="27"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs>
            <image
              height="27"
              href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAB4AAAAeCAMAAAAM7l6QAAAAAXNSR0IB2cksfwAAARRQTFRFAAAASk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pVwjv/QAAAFx0Uk5TADWDv+P19ufLh0YFAyuM2vr/++GfOQJf1/7qUFvYWDIvTHX590sBMLQ8IKruPhG34pmlTdMWLOvoKRnzMxg0JuUbi6gE0E61/Mx5VBKakwv9DT+QsQeSdwgPsNTBV9X7AAABhUlEQVR4nHWS61aCQBSF8RokBhutxJq0TNRMK6+V3bQ0ukMqqb3/ezQ1BjhrsX/tMx+Lc+bsEQSmUDgSjcXXRGk9IfCSkxuKij9pqfTmFoe3Mzo8ZXdWf7Cr0UMi7eXy+wdiATgM+7i8TSkpGqwySmUdmiS7OJkBxIrbzziqAsc1F2/oIBV/rzoBTv6LkAIUV2ctnkIPLX1YBTFWL2KcAY1l9wgg8Xtoqmi1mY0CezzudHF+wWwMyPH4kqB3xWwcyPP4OoPeDbNrQJPHt3foJpkVgQMe1+JQ+sxKGkT+YgPg/oH59RQKpVVeGkIfLX0iDTwe+elW2cSTm8lmFnq17qPPOswXr9w5pIHu+wMFefU+T4Rp4NG3Zufy+rY2GJr0bajvHz4uHf8edUnmji4JJlFhpW2Py58n7mPTn15e3y1YY984NPZG67zX6yr3IzrzR9qCNpn6udy+uLpJ9tk27LEGy7GFINUnFtSvWSCfOnQ+Zx7I7S8L5iIQCzPHhBqMhflC/f4BU6Uw/3BpkC8AAAAASUVORK5CYII="
              id="search-icon"
              width="27"
            />
          </defs>
          <use
            href="#search-icon"
            x="0"
            y="0"
          />
        </svg>
      </button>
    </form>
    <div
      className="sc-fzokOt cLyWOW"
    >
      <h3
        className="sc-fzqBZW dCxikg"
      >
        Recommended ingredients
      </h3>
      <div
        className="sc-fzqNJr dMSFGw"
      />
    </div>
    <div
      className="sc-fznMAR dHZGIM"
    >
      <div
        className="sc-fznJRM ZeGtB"
      >
        <button
          className="sc-AxjAm ewqDcP sc-AxirZ sc-fznWqX updjA"
          onClick={[Function]}
          type="button"
        >
          <svg
            height="20"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <defs>
              <image
                height="20"
                href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADIAAAAyCAMAAAAp4XiDAAAAAXNSR0IB2cksfwAAAQ5QTFRF8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unk8unkQ40o0AAAAFp0Uk5TADnX8/KwE4j/9lWN+VoshozL/LSLSeL90vcbYma6+5xlMQZ6sbOTEYX1UjyEHp7GB0YUh7lZKMzg/jWsz1Mrd3lqGkKOFtQPvtwm378ELzPmMg7h7Nn62CW1Gbu+JwAAAQ5JREFUeJxjYCAMGJmYQYCFlY0IxRDAzgEBnFxEa+GGauHhHX5a+PgFCAFBqBYhYSBHRJRBjINEIM4gQaoWSQYpaRlCQBaqWE4eyFFQHLyBPHBalJRVVEFATR3M1YBq0dTCrUVbHKpIB8zV1dMHA2UD3FoMYdGmRrTjh6oWIy5eDGAM02KCKWdqxmBOauJnZrAgVYslgxU7Zs61hknbYMqJ2A7eQKanFjt7BxBwdALznGFaXHDrcHWDqrEGc909IBnG0wu3Fi9vqBYfoh0/TLT4+vn7+2sFQLWY+xMCgUEMKqQm/mCGEFK1cDCYSoSGhnqHQbmaoYRAOPvgDWQ6aomAafEgWotBJESHJT8RigH0OlVdPf84OAAAAABJRU5ErkJggg=="
                id="filter-icon"
                width="20"
              />
            </defs>
            <use
              href="#filter-icon"
              x="0"
              y="0"
            />
          </svg>
          Filter
        </button>
        <div />
      </div>
      
      <div
        className="sc-fzolEj koFjRo"
      >
        No recipes found
      </div>
    </div>
  </div>
</div>
`;
